### PR TITLE
adding coverage using mutation testing for mutants 3, 7 and alon's rule

### PR DESCRIPTION
**Pull request by MichaelMorami**

### CODE REVIEW

File: certora/specs/ghoVariableDebtToken_summarized.spec
Comment (gadicer) [inline certora/specs/ghoVariableDebtToken_summarized.spec:1]: updateDiscountDistribution calls _refreshDiscountPercent twice. So refresh_called  could flip twice. Better use a counter.

File: certora/specs/ghoVariableDebtToken.spec [hunk 1]
```diff
@@ -128,6 +130,17 @@ invariant discountCantExceed100Percent(address user)
...
 	}
+    
+/**
+* @title at any point in time, the user's discount rate isn't larger than DISCOUNT_RATE
+**/
+invariant discountCantExceedDiscountRate(address user)
+	getUserDiscountRate(user) <= discStrategy.DISCOUNT_RATE()
+	{
+		preserved updateDiscountDistribution(address sender,address recipient,uint256 senderDiscountTokenBalance,uint256 recipientDiscountTokenBalance,uint256 amount) with (env e) {
+			require(indexAtTimestamp(e.block.timestamp) >= ray());
```
Comment (gadicer) [inline certora/specs/ghoVariableDebtToken.spec:145]: function indexAtTimestamp has already the same require. This require should be redundant. Does the invariant fail without it?

File: certora/specs/ghoVariableDebtToken.spec [hunk 2]
```diff
@@ -485,17 +498,17 @@ rule integrityOfBurn_userIsolation() {
 * Integrity of updateDiscountDistribution
 ***************************************************************/
 
-/**
-* @title proves that the discount rate is calculated correctly when calling updateDiscountDistribution
-**/
+// /**
+// * @title proves that the discount rate is calculated correctly when calling updateDiscountDistribution
+// **/
```
Comment (gadicer) [inline certora/specs/ghoVariableDebtToken.spec:507]: Does the rule finish successfully? If it times out it should be excluded from CI?

File: certora/specs/ghoVariableDebtToken.spec [hunk 3]
```diff
@@ -128,6 +130,17 @@ invariant discountCantExceed100Percent(address user)
...
+    
+/**
+* @title at any point in time, the user's discount rate isn't larger than DISCOUNT_RATE
+**/
+invariant discountCantExceedDiscountRate(address user)
+	getUserDiscountRate(user) <= discStrategy.DISCOUNT_RATE()
+	{
+		preserved updateDiscountDistribution(address sender,address recipient,uint256 senderDiscountTokenBalance,uint256 recipientDiscountTokenBalance,uint256 amount) with (env e) {
+			require(indexAtTimestamp(e.block.timestamp) >= ray());
+		}
```
Comment (gadicer) [inline certora/specs/ghoVariableDebtToken.spec:146]: rule accumulated_interest_increase_after_mint can be simplified using this invariant.

File: certora/specs/ghoVariableDebtToken_summarized.spec
Comment (MichaelMorami) [reply certora/specs/ghoVariableDebtToken_summarized.spec:1]: done

File: certora/specs/ghoVariableDebtToken.spec (hunk 1 above)
Comment (MichaelMorami) [reply certora/specs/ghoVariableDebtToken.spec:145]: removed

File: certora/specs/ghoVariableDebtToken.spec (hunk 2 above)
Comment (MichaelMorami) [reply certora/specs/ghoVariableDebtToken.spec:507]: it's commented out and not included in CI

File: certora/specs/ghoVariableDebtToken.spec (hunk 3 above)
Comment (MichaelMorami) [reply certora/specs/ghoVariableDebtToken.spec:146]: then you can use it once i merge

**Pull request merged**